barcode/templates/barcode/baseextra.html

```diff
@@ -0,0 +1,98 @@
+{% extends "baseextra.html" %}
+{{ super() }}
+
+{% if detect_barcode_scan %}
+  <script type="text/javascript">
+      var getBadgeNumFromBarcode = function (barcode, callback) {
+          barcode = ('' + barcode).trim();
+          callback = callback || function () {
+              };
+
+          var regexpUUID = /^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}$/i;
+
+          // Remove non-word characters bracketing the barcode if they match
+          // Some scanners add these but we don't want them
+          var first = barcode.substr(0, 1),
+              last = barcode.substr(barcode.length - 1);
+          if (first === last && last.match(/\W/g)) {
+              barcode = barcode.substr(1, barcode.length - 2);
+          }
+
+          // Bypass decryption if it's a number or a UUID
+          if ($.isNumeric(barcode) || barcode.match(regexpUUID)) {
+              callback({'barcode': barcode});
+              return;
+          }
+
+          $.ajax({
+              method: 'POST',
+              url: '../barcode/get_badge_num_from_barcode',
+              data: {
+                  barcode: barcode,
+                  csrf_token: csrf_token
+              },
+              success: function (response, status) {
+                  if (response && response['badge_num'] === -1) {
+                      response['error'] = response['message'];
+                  }
+                  callback(response);
+              },
+              error: function (response, status, statusText) {
+                  callback({'error': 'Could not decode barcode: ' + statusText});
+              }
+          });
+      };
+
+      // The currently-selected field is used as the default place to put a barcode
+      var $barcodeField = $(document.activeElement);
+
+      // Override these per-page as necessary
+      var setBarcodeField = function () {
+      };
+
+      var useBarcode = function (barcode) {
+      };
+
+      var scannerOnReceive = function (event) {
+          if (event.key === '~' || event.key === '\\') {
+              event.preventDefault();
+              $barcodeField.blur();
+          }
+      };
+
+      var scannerOnError = function () {
+          $barcodeField.focus();
+      };
+
+      var scannerOnComplete = function (barcode, qty) {
+          getBadgeNumFromBarcode(barcode, function (response) {
+              $barcodeField.focus();
+              if (response) {
+                  if (response['error']) {
+                      toastr.error(response['error'], '', {timeOut: 3000});
+                  } else if (response['barcode']) {
+                      if ($barcodeField.is('input[name=badge_num]')) {
```

Comment:
This is now a more sane 'default', only applying the barcode to a field if it's a field with the name `badge_num`. I almost took this out as well, but then I found myself placing it in overrides for every page that uses this code, so I left it here.